install.sh

```diff
@@ -35,16 +42,90 @@ fi
-# Move binaries to their correct locations
-echo "$EMOJI Installing roller..."
-sudo mv "/tmp/roller_tmp/roller_bins/lib"/* "$INTERNAL_DIR"
-sudo mv "/tmp/roller_tmp/roller_bins/roller" "$ROLLER_BIN_PATH"
-sudo mv "/tmp/roller_tmp/roller_bins/rollapp_evm" "$ROLLAPP_EVM_PATH"  # move the rollapp_evm binary
+if ! sudo curl -L "$TGZ_URL" --progress-bar | sudo tar -xz -C "/tmp/roller_tmp"; then
+    # Curl failed. Fallback to clone and build.
+    echo "$EMOJI Download failed. Cloning and building manually..."
+    # Check prerequisites
+    command -v curl >/dev/null 2>&1 || { echo >&2 "$EMOJI curl is required but it's not installed. Aborting."; exit 1; }
+    command -v wget >/dev/null 2>&1 || { echo >&2 "$EMOJI wget is required but it's not installed. Aborting."; exit 1; }
+    command -v git >/dev/null 2>&1 || { echo >&2 "$EMOJI git is required but it's not installed. Aborting."; exit 1; }
+    if ! command -v go >/dev/null 2>&1; then
+        echo >&2 "$EMOJI Go is required but it's not installed. Aborting."
+        display_go_installation_instructions
+        exit 1;
+    fi
+
+    # Check go version
+    GO_VERSION=$(go version | awk -F' ' '{print substr($3, 3, 4)}')  # Extract major.minor version number (remove the 'go' prefix)
+    if [ "$GO_VERSION" != "$GO_REQUIRED_VERSION" ]; then
+        echo "$EMOJI Go version $GO_REQUIRED_VERSION is required, but you have version $GO_VERSION. Aborting."
+        display_go_installation_instructions
+        exit 1
+    fi
+
+    for i in "${!PROJECTS[@]}"; do
+        echo "$EMOJI handling ${PROJECTS[i]}..."
+        cd /tmp
+
+        # Check if binaries already exist
+        IFS=' ' read -r -a binary_array <<< "${BINARYNAME[i]}"  # convert string to array
+        binary=${binary_array[0]}  # get the first element
+        echo "$EMOJI checking for $binary..."
+        if [ -x "$(command -v "$binary")" ]; then
+            read -p "$binary already exists. Do you want to overwrite it? (y/n) " -n 1 -r
+            echo
+            if [[ $REPLY =~ ^[Nn]$ ]]; then
+                binary_path=$(which "$binary" 2> /dev/null)  # Redirect error output to /dev/null
+                sudo cp "$binary_path" "$INTERNAL_DIR"
+                continue 1
+            else
+                rm -f "$binary_path"
+            fi
+        fi
+
+        REPO_URL=${REPOS[i]:-"https://github.com/dymensionxyz/${PROJECTS[i]}"}
+        if [ ! -d "/tmp/${PROJECTS[i]}" ]; then
+            git clone "$REPO_URL"
+        fi
+        cd "${PROJECTS[i]}"
+        if [ -n "${VERSIONS[i]}" ]; then
+            git checkout "${VERSIONS[i]}"
+        fi
+        if [ -n "${BUILDCOMMANDS[i]}" ]; then
+            ${BUILDCOMMANDS[i]}
+        else
+            make install
+        fi
+
+        for binary in ${BINARYNAME[i]}; do
+            binary_path=$(which "$binary" 2> /dev/null)  # Redirect error output to /dev/null
```

Comment:
problematic again. the user may have both the standard and the dymension relayer on the same machine

Reply:
in this case the user had the option to use the existing one or overwrite